packages/controllers/src/snaps/utils.test.ts

```diff
@@ -122,3 +126,92 @@ describe('getSnapPrefix', () => {
     );
   });
 });
+
+describe('isValidSnapVersionRange', () => {
+  it('supports *', () => {
+    expect(isValidSnapVersionRange('*')).toBe(true);
+  });
+
+  it('supports normal version ranges', () => {
+    expect(isValidSnapVersionRange('^1.2.3')).toBe(true);
+    expect(isValidSnapVersionRange('1.5.0')).toBe(true);
+  });
+
+  it('supports pre-release versions', () => {
+    expect(isValidSnapVersionRange('1.0.0-beta.1')).toBe(true);
+    expect(isValidSnapVersionRange('^1.0.0-beta.1')).toBe(true);
+  });
+
+  it('rejects non strings', () => {
+    expect(isValidSnapVersionRange(null)).toBe(false);
+    expect(isValidSnapVersionRange(undefined)).toBe(false);
+    expect(isValidSnapVersionRange(2)).toBe(false);
+    expect(isValidSnapVersionRange(true)).toBe(false);
+    expect(isValidSnapVersionRange({})).toBe(false);
+  });
+});
+
+describe('gtVersion', () => {
+  it('supports regular versions', () => {
+    expect(gtVersion('1.2.3', '1.0.0')).toBe(true);
+    expect(gtVersion('2.0.0', '1.0.0')).toBe(true);
+    expect(gtVersion('1.0.0', '1.2.3')).toBe(false);
+    expect(gtVersion('1.0.0', '2.0.0')).toBe(false);
+  });
+
+  it('supports pre-release versions', () => {
+    expect(gtVersion('1.0.0-beta.2', '1.0.0-beta.1')).toBe(true);
+    expect(gtVersion('1.0.0-beta.2', '1.2.3')).toBe(false);
+    expect(gtVersion('1.0.0', '1.0.0-beta.2')).toBe(true);
+    expect(gtVersion('1.2.3-beta.1', '1.0.0')).toBe(true);
```

Comment:
what happens in `gtVersion('1.2.3-beta.1', '1.2.3-alpha.2')`?

Reply:
`true`